easybuild/tools/module_generator.py

```diff
@@ -1152,12 +1161,13 @@ class ModuleGeneratorLua(ModuleGenerator):
     LOAD_TEMPLATE_DEPENDS_ON = 'depends_on("%(mod_name)s")'
     IS_LOADED_TEMPLATE = 'isloaded("%s")'
 
-    PATH_JOIN_TEMPLATE = 'pathJoin(root, "%s")'
+    PATH_JOIN_TEMPLATE = 'pathJoin(root, "%s")'  # TODO: remove after 6.0, replaced by _path_join_cmd()
```

Comment:
If we really want to deprecate this, we should make it a class property that emit a deprecation warning via `self.log.deprecated`?